### ISSUE TITLE
"Recombinant" or "Recombination node"

### ISSUE THREAD

**Issue by hyanwong** [jeromekelleher/sc2ts#115]
I'm going through the codebase and the manuscript, and I see we use the word "recombinant" in lots of places (e.g. class names, variable names). I suggest that most people will think of a recombinant as a sample node that has recombination in its ancestry, e.g. "I've been infected by a recombinant (virus)". So I wonder if we want to change the terminology in the code to refer to the nodes in which recombination events actually took place as "recombination nodes" instead? E.g. instead of the `Recombinant` class, we should have a `RecombinationNode` class.

It's a bit of a large change, but should be reasonably easy to make with auto search-and-replace I think. Thoughts?

**Comment by szhan** [comment]
I have a (very) slight preference for `recombinant nodes`, because these nodes represent recombinant sequences (sampled or reconstructed) or products of recombination events.

**Comment by hyanwong** [comment]
I'm still uneasy about this, TBH. E.g. in this plot:

![image_720](https://user-images.githubusercontent.com/4699014/229774930-1af95f11-a035-4311-843b-cf09da819832.png)

I think of the black node as a "recombination node", not a "recombinant". I think of the immediate children of the black node as recombinants. I also suspect that lots of researchers would think of all the purple and blue nodes as "recombinant viruses"?

Could we see what other people on the project think, perhaps?

**Comment by jeromekelleher** [comment]
We're splitting hairs here - the recombination node represents the original recombinant sequence. The children of a recombinant aren't necessarily recombinants, they may be later descendants of that sequence. We only insert recombination nodes as a convenience to make it easy to find what we think are unique recombinants - we may have multiple sequences from the same day which have the same copying path and we therefore "path compress" these together into a single node, representing the unique recombinant origin.

**Comment by jeromekelleher** [comment]
There is the distinction of the "causal" sequences for a recombinant/recombination node. These are the sequences that required a path of > 1 parent, resulting in the creation of the recombination node, which represents the original recombinant sequence.

**Comment by hyanwong** [comment]
> Also, talking about "recombination nodes" is confusing because it ties to that specific ARG. We want to be able to refer to the original recombinant sequence that was the result of recombination between two strains within a host. Using "node" to refer to this sequence is misleading.

I agree that the "recombination node" is tied to a specific ARG, and a recombinant is a sequence that results directly from a recombination. But I often *want* to refer to the specific (essentially, inferred) node in the specific ARG when e.g. describing a plot like the one above. I think labelling the black dot a "recombination node" here is less confusing, as it is deliberately tied to this particular ARG.
